aten/src/ATen/native/PixelShuffle.cpp

```diff
@@ -129,7 +129,8 @@ Tensor math_pixel_shuffle(const Tensor& self, int64_t upscale_factor) {
   std::vector<int64_t> final_shape(self.sizes().begin(), self_sizes_batch_end);
   final_shape.insert(final_shape.end(), {oc, oh, ow});
 
-  return input_permuted.reshape(final_shape);
+  // pixel_shuffle expects to *never* return an alias of the input.
+  return input_permuted.reshape(final_shape).clone();
```

Comment:
Just do: `input_permuted.clone().view(final_shape)` to avoid the extra copy ;)

Reply:
oh you're totally right haha